### PR TITLE
Sentinel-1 false color composite

### PR DESCRIPTION
**Pull request by mitkin**
PR for Sentinel false color composite.

There are some issues I experience now: 
- can't load both polarizations at once, thus loading it as two scenes. There must be a better way!

### CODE REVIEW

**Comment by djhoese** [comment]
When things get "complicated" like this you have to go an extra step to specify what you want. I believe when you use the keywords during loading the way you have that it finds the first and "best" match it can find. To get what you want, you could try:

```
from satpy import DatasetID
scn.load([DatasetID(name='measurement', polarization='hh'), DatasetID(name='measurement', polarization='hv')])
```

You may want to create these IDs before so that you can use them to retrieve the datasets later `scn[ds1]`. Could you also please change your Scene import to `from satpy import Scene`. I'd like the most generic versions of the imports to be used in examples.

**Comment by mitkin** [comment]
@davidh-ssec ok, thanks! Specifying datasets explicitly works fine.

**Comment by mitkin** [comment]
@mraspaud @davidh-ssec ready to merge?

File: satpy/sentinel1-false-color.ipynb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    "    )\n",
+    "\n",
+    "    composite = 'hv'\n",
+    "\n",
+    "    scn.load([DatasetID(name='measurement', polarization='hh'),\n",
+    "              DatasetID(name='measurement', polarization='hv')])\n",
+    "\n",
+    "    mx = 2000.\n",
+    "\n",
+    "    i = scn.datasets.values()[0]\n",
```
Comment (mraspaud) [inline satpy/sentinel1-false-color.ipynb]: Would be nice to have explicitly which dataset is used.
What about using eg `i = scn[DatasetId(name='measurement', polarization='hh')]`

File: satpy/sentinel1-false-color.ipynb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    "    m = scn.datasets.values()[1]\n",
+    "\n",
+    "    rs = mx - (mx - m) * (mx - i) / mx\n",
+    "    data_g = ((mx - i) * m + rs) * i / mx\n",
+    "\n",
```
Comment (mraspaud) [inline satpy/sentinel1-false-color.ipynb]: Have you compared with the building `SARIce` composite ?

Comment (mitkin) [reply satpy/sentinel1-false-color.ipynb]: Nope, I looked up the docs but it's not very obvious how to generate a composite and retrieve an image for post processing. How would you do it?

File: satpy/sentinel1-false-color.ipynb (hunk 1 above)
Comment (mitkin) [reply satpy/sentinel1-false-color.ipynb]: ok

File: satpy/sentinel1-false-color.ipynb (hunk 2 above)
Comment (mraspaud) [reply satpy/sentinel1-false-color.ipynb]: To generate the composite, just load it: `scn.load['sarice']`
To do your own post-processing, just use `get_enhanced_image` : http://satpy.readthedocs.io/en/latest/satpy.writers.html?#satpy.writers.get_enhanced_image

Comment (mraspaud) [reply satpy/sentinel1-false-color.ipynb]: the enhancements you can put in the `etc/enhancements/generic.yaml` file

Comment (mitkin) [reply satpy/sentinel1-false-color.ipynb]: Strange, I am not able to save the composite dataset. When `scn.save_dataset('sar-ice', 'a1.png')` I get a RunTimeError `RuntimeError: No enhancement configuration files found or specified, cannot automatically enhance dataset`. Looks like I can't use `writer.save_dataset` without specifying the enhancer. It doesn't use the existing enhancer `sar-ice` for some reason. I guess I stick to what it is now, and leave the enhancements for the other example.

Comment (mraspaud) [reply satpy/sentinel1-false-color.ipynb]: sounds good

**Comment by djhoese** [comment]
I know @mraspaud merged this already, but maybe we should consider making all examples use `reader` and `filenames` keyword arguments to the Scene instead of the "discovery" keyword arguments. Especially since the latter will be separate in to its own function in the near future.

**Comment by mitkin** [comment]
@davidh-ssec , yeah, that would be my preference actually. This is exactly what I wanted to put in the example, but reading the data like `Scene(reader='safe_sar_c', filenames=filelist)` where a single file name in the filelist is the full path to a folder like `S1A_EW_GRDM_1SDH_20170830T174814_20170830T174914_018156_01E80F_29CF.SAFE` will not find supported files. I wish I had more time to dig into it right now, but it will have to wait a little.

**Pull request merged**